client/src/main/java/io/pravega/client/stream/impl/ModelHelper.java

```diff
@@ -351,6 +351,9 @@ public static final SegmentRange createSegmentRange(final String scope, final St
                                                                                        final List<SegmentId> segments, String delegationToken) {
         Exceptions.checkNotNullOrEmpty(scope, "scope");
         Exceptions.checkNotNullOrEmpty(stream, "stream");
+        Exceptions.checkArgument(segments.stream().allMatch(x -> x.getStreamInfo().getScope().equals(scope) &&
+                        x.getStreamInfo().getStream().equals(stream)),
+                "streamInfo", "stream info does not match segment id", scope, stream, segments);
```

Comment:
that may require doing it outside of checkArgument to maintain the exact segment id(s) so that we can log it. 
instead, here we print all the segments so that one can see which ones are invalid.